overlord/ifacestate/helpers.go

```diff
@@ -321,6 +321,30 @@ func (m *InterfaceManager) reloadConnections(snapName string) ([]string, error)
 		return nil, err
 	}
 
+	canReload := false
+	var policyChecker interfaces.PolicyFunc
+	var autoChecker *autoConnectChecker
+	var connChecker *connectChecker
+
+	deviceCtx, err := snapstate.DeviceCtx(m.state, nil, nil)
+	if errors.Is(err, state.ErrNoState) {
+		//noop
```

Comment:
this comment should be probably be expanded: // if there is no model there should be no connections so the rest is a noop anyway

Reply:
I'll update the comment, but right now this just skips the "reloading static attrs part" but could it not just skip the entire rest of the function in that case? i.e. could it return `nil, nil`?

Reply:
I played with this a little bit more, and I actually think that this is a dead code path. I think the `getConns` call immediately above will return `errNoState` if it's an issue, so this new stuff isn't actually introducing any new dependencies or constraints into the logic of this function, which is what I was originally trying to avoid